### PR TITLE
Update & Improve Readme

### PR DESCRIPTION
**Pull request by johannes-wolf**

### CODE REVIEW

**Comment by josephbirkner** [comment]
Nice 🎇 Some more ideas:

* Graphical Architecture Overview (Relationships between `Expr`, `Environment`, `Fields`, `Model`, `Nodes`)
* API integration example with CMake/C++
* C++ Model extension example?

**Comment by johannes-wolf** [comment]
Would like to merge this and add additions as separate PRs.

File: README.md [hunk 1]
```diff
@@ -4,4 +4,142 @@
...
 
 Simfil is a language for querying tree structured data.
+
+## Motivation
+- **Simplicity**: the language should be very simple but yet powerful to allow for complex queries.
+- **Speed**: querying models should be fast and scale to multiple cores.
+- **Efficiency**: the internal model should be memory efficient, as simfil is designed to query huge amounts of data.
+
+## Query Language
 For details about the language see the [Language Guide](simfil-language.md).
```
Comment (MisterGC) [inline README.md:14]: Which kind of sources of inspiration have been used to come up with the language, what are the key aspects covered - as we list specific examples we should also briefly cover the language (and then still reference the language specification)

File: README.md [hunk 2]
```diff
@@ -4,4 +4,142 @@
...
+Some types have special operators. The type `irange` supports the unpack operator `...` to expand its list of values.
+```
+range(1,25)...{count((_ % range(1,_)...) == 0) == 2}
+```
+```
+2, 3, 5, 7, 11, 13, 17, 19, 23
+```
+
+## Building the Project
+```sh
```
Comment (MisterGC) [inline README.md:67]: Write something about the platforms/toolchains that have been confirmed/can be used - are there specific requirements/dependencies?

File: README.md [hunk 3]
```diff
@@ -4,4 +4,142 @@
...
+mkdir build && cd build
+cmake .. && cmake --build .
+ctest
+```
+
+## Using the Repl
```
Comment (MisterGC) [inline README.md:73]: Don't assume that all users know 'Read-Eval-Print-Loop' but call the section something like 'Try it yourself with the simfil command line tool'  or something. Describe what the tool is about.

File: README.md [hunk 4]
```diff
@@ -4,4 +4,142 @@
 [![](https://gist.githubusercontent.com/johannes-wolf/61e57af50757b03e0c7cd119ec2d2f4b/raw/0ae49c7509dea18b4c110b8bf416f2715a214933/badge.svg)](https://github.com/Klebert-Engineering/simfil)
 
 Simfil is a language for querying tree structured data.
+
+## Motivation
+- **Simplicity**: the language should be very simple but yet powerful to allow for complex queries.
+- **Speed**: querying models should be fast and scale to multiple cores.
+- **Efficiency**: the internal model should be memory efficient, as simfil is designed to query huge amounts of data.
+
+## Query Language
```
Comment (MisterGC) [inline README.md:13]: Add a 'Repo Structure' section and briefly describe about how the repo is organized.

**Review by MisterGC — CHANGES_REQUESTED** [review]

**Review by MisterGC — APPROVED** [review]

**Comment by johannes-wolf** [comment]
Will do additional changes as follow up PRs.

**Pull request merged**